pkg/platform/kube/functionres/lazy.go

```diff
@@ -1547,13 +1568,15 @@ func (lc *lazyClient) generateCronTriggerCronJobSpec(functionLabels labels.Set,
 		eventBodyFilePath := "/tmp/eventbody.out"
 		eventBodyCurlArg := fmt.Sprintf("--data '@%s'", eventBodyFilePath)
 
-		// if body is a valid JSON parse it accordingly
-		eventBodyAsJSON, err := json.Marshal(eventBody)
-		if err == nil {
-			eventBody = string(eventBodyAsJSON)
+		// try compact as JSON (will fail if it's not a valid JSON)
+		eventBodyAsCompactedJSON := bytes.NewBuffer([]byte{})
+		if err := json.Compact(eventBodyAsCompactedJSON, []byte(eventBody)); err == nil {
+
+			// set the compacted JSON as event body
+			eventBody = eventBodyAsCompactedJSON.String()
 		}
 
-		curlCommand = fmt.Sprintf("echo %s > %s && %s %s",
+		curlCommand = fmt.Sprintf("echo '%s' > %s && %s %s",
```

Comment:
using go builtin `strconv.quoute(data)` can provide you such functionality of escaping/quoting data because imo, if your function config is

```
meta:
  annotation:
    something: this is just a 'test'
spec:
  ...
```

then the `'` will break quoting here.